File: src/za/co/adhd_developers/java/questions/Question44.java
package za.co.adhd_developers.java.questions;

import za.co.adhd_developers.java.tools.PolygonalUtil;
import za.co.adhd_developers.java.tools.Question;

public class Question44 implements Question {

    private int answer = 0;

    @Override
    public void doWork() {
        Boolean keepGoing = true;
        int currNumber = 2;

        while (keepGoing) {
            int currPentagonal = (int) PolygonalUtil.getPentagonal(currNumber);

            for (int j = 1; j < currNumber; j++) {
                int otherPentagonal = (int) PolygonalUtil.getPentagonal(j);

                if (PolygonalUtil.isPentagonal(currPentagonal - otherPentagonal)) {
                    if (PolygonalUtil.isPentagonal(currPentagonal + otherPentagonal)) {
                        this.answer = currPentagonal - otherPentagonal;

                        keepGoing = false;
                        break;
                    }
                }
            }

            currNumber++;
        }
    }

    @Override
    public void printAnswer() {
        System.out.println("=================================");
        System.out.println("Question 44");
        System.out.println("The difference of the pentagonal pair that matches the criteria, is: " + this.answer);
        System.out.println("=================================");
    }
}

File: src/za/co/adhd_developers/kotlin/questions/Question44.kt
package za.co.adhd_developers.kotlin.questions

import za.co.adhd_developers.kotlin.Question
import za.co.adhd_developers.kotlin.tools.MyToolbox

class Question44 : Question {

    var answer = 0.0

    override fun doWork() {
        for (j in 1..10000) {
            val pJ = MyToolbox.PolygonalToolBox.getPentagonal(j)

            for (k in j + 1..10000) {
                val pK = MyToolbox.PolygonalToolBox.getPentagonal(k)

                if (MyToolbox.PolygonalToolBox.isPentagonal(pJ.plus(pK))) {
                    if (MyToolbox.PolygonalToolBox.isPentagonal(pK.minus(pJ))) {
                        this.answer = pK.minus(pJ)
                    }
                }
            }
        }
    }

    override fun printAnswer() {
        println("=================================")
        println("Question 44")
        println("The difference of th pentagonal pair that matches the criteria, is: " + this.answer.toInt())
        println("=================================")
    }
}